tempest/api/workloadmgr/security_groups/test_security_group_default.py

```
@@ -0,0 +1,322 @@
+                workload_id = self._create_workload([self.vm_id])
+                LOG.debug("\nWorkload created : {}\n".format(workload_id))
+
+                # Take a Full snapshot
+                LOG.debug("Take a full snapshot")
+                snapshot_id = self._take_full_snapshot(workload_id)
+
+                # Get the sec groups and rules list for verification post restore
+                secgroups_before = self.list_security_groups()
+                rules_before = self.list_security_group_rules()
+
+                # Delete vms, volumes, security groups
+                try:
+                    LOG.debug("Delete VM + volume + security groups")
+                    self.delete_vm_secgroups([self.vm_id], created_security_groups)
+                    reporting.add_test_step(
+                        "Delete vm and security group", tvaultconf.PASS
+                    )
+                except Exception as e:
+                    LOG.error(f"Exception: {e}")
+                    raise Exception("Deletion of vm and security group/s failed")
+                time.sleep(60)
+
+                # Perform security groups restore through wlm command
+                self._restore_through_wlm_cli(snapshot_id)
+
+                # Security group and rules verification post restore
+                LOG.debug("Compare the security groups before and after restore")
+                secgroups_after = self.list_security_groups()
+                if len(secgroups_after) == len(secgroups_before):
+                    reporting.add_test_step(
+                        "Security group verification pre and post restore",
+                        tvaultconf.PASS,
+                    )
+                else:
+                    LOG.error("Security groups verification pre and post restore")
+                    reporting.add_test_step(
+                        "Security groups verification failed pre and post restore",
+                        tvaultconf.FAIL,
```

test script status also need to be marked as FAIL explicitly

---

done